amr-wind/physics/SyntheticTurbulence.cpp

```diff
@@ -0,0 +1,574 @@
+                                                           umin, umax));
+  } else {
+    throw std::runtime_error("SyntheticTurbulence: invalid mean wind type specified = " + mean_wind_type);
+  }
+
+  // Smearing factors
+  pp.query("grid_spacing", m_grid_spacing);
+  m_epsilon = 2.0 * m_grid_spacing;
+  pp.query("gauss_smearing_factor", m_epsilon);
+  m_gauss_scaling = 1.0 / (m_epsilon * std::sqrt(pi));
+
+  // Time offsets if any...
+  pp.query("time_offset", m_time_offset);
+
+  // Done reading user inputs, process derived data
+
+  // Center of the grid
+  m_turb_grid.origin[0] = location[0];
+  m_turb_grid.origin[1] = location[1];
+  m_turb_grid.origin[2] = location[2];
+
+  // Compute box-fixed reference frame.
+  //
+  // x-direction points to flow direction (convert from compass direction to vector)
+  m_turb_grid.tr_mat.xx() = -std::sin(wind_direction);
+  m_turb_grid.tr_mat.xy() = -std::cos(wind_direction);
+  m_turb_grid.tr_mat.xz() = 0.0;
+  // z always points upwards (for now...)
+  m_turb_grid.tr_mat.zx() = 0.0;
+  m_turb_grid.tr_mat.zy() = 0.0;
+  m_turb_grid.tr_mat.zz() = 1.0;
+  // y = z .cross. x
+  m_turb_grid.tr_mat.y() = m_turb_grid.tr_mat.z() ^ m_turb_grid.tr_mat.x();
+  //utils::cross_prod(&m_turb_grid.tr_mat[2][0], &m_turb_grid.tr_mat[0][0], &m_turb_grid.tr_mat[1][0]);
+
+  amrex::Print()
+    << "Synthethic turbulence forcing initialized \n"
+    << "  Turbulence file = " << m_turb_filename << "\n"
+    << "  Box lengths = [" << m_turb_grid.box_len[0] << ", "
+    << m_turb_grid.box_len[1] << ", " << m_turb_grid.box_len[2] << "]\n"
+    << "  Box dims = [" << m_turb_grid.box_dims[0] << ", "
+    << m_turb_grid.box_dims[1] << ", " << m_turb_grid.box_dims[2] << "]\n"
+    << "  Grid dx = [" << m_turb_grid.dx[0] << ", " << m_turb_grid.dx[1] << ", "
+    << m_turb_grid.dx[2] << "]\n"
+    << "  Centroid (forcing plane) = [" << m_turb_grid.origin[0] << ", "
+    << m_turb_grid.origin[1] << ", " << m_turb_grid.origin[2] << "]\n"
+    << "  Mean wind profile: U = " << m_wind_profile->reference_velocity()
+    << " m/s; Dir = " << wind_direction * 180.0 / pi
+    << " deg; type = " << mean_wind_type << std::endl;
+}
+
+void SyntheticTurbulence::initialize_fields(
+    int // level
+    ,
+    const amrex::Geometry& // geom)
+)
+{
+    //TODO: Figure out what goes here
+}
+
+void SyntheticTurbulence::pre_advance_work()
+{
+  if (m_is_init)
+    initialize();
+
+  update();
+}
+
+void SyntheticTurbulence::initialize()
+{
+  // Convert current time to an equivalent length based on the reference
+  // velocity to determine the position within the turbulence grid
+  const amrex::Real curTime = m_time.new_time() - m_time_offset;
+  const amrex::Real eqivLen = m_wind_profile->reference_velocity() * curTime;
+  int il, ir;
+  get_lr_indices(m_turb_grid, 0, eqivLen, il, ir);
+  load_turb_plane_data(m_turb_filename, m_turb_grid, il, ir);
+
+  m_is_init = false;
+}
+
+void SyntheticTurbulence::update()
```

Comment:
Make this member public in header

Reply:
Fixed in https://github.com/Exawind/amr-wind/pull/327/commits/5213aab7053bb736e11a39c02ff753c2820f8837